fe/fe-core/src/main/java/com/starrocks/authentication/PlainPasswordAuthenticationProvider.java

```diff
@@ -75,4 +76,15 @@ public void authenticate(
             throw new AuthenticationException("password mismatch!");
         }
     }
+
+    @Override
+    public UserAuthenticationInfo upgradedFromPassword(UserIdentity userIdentity, Password password)
+            throws AuthenticationException {
+        UserAuthenticationInfo ret = new UserAuthenticationInfo();
+        ret.setPassword(password.getPassword());
+        ret.setAuthPlugin(PLUGIN_NAME);
+        ret.setOrigUserHost(userIdentity.getQualifiedUser(), userIdentity.getHost());
+        ret.setTextForAuthPlugin(password.getUserForAuthPlugin());
```

Comment:
what does `userForAuthPlugin` mean?

Reply:
actually for plain password this field is empty, but for other authentication plugins such as kerberos, this field normally is used for additional information.